experiments/kompanion/cmd/export/export.go

```diff
@@ -141,66 +248,60 @@ func run(_ *cobra.Command, _ []string) {
 	sort.Slice(resources, func(i, j int) bool {
 		return resources[i].SingularName < resources[j].SingularName
 	})
+
+	// todo acpana debug logs
 	// log.Printf("Going to iterate over the following resources %+v", resourcesToName(resources))
 
 	namespaces, err := clientset.CoreV1().Namespaces().List(context.TODO(), metav1.ListOptions{})
 	if err != nil {
 		log.Fatalf("Error fetching namespaces: %s\n", err)
 	}
 
-	reportName := timestampedName("report")
+	reportName = timestampedName("report")
 	if err := os.Mkdir(filepath.Join(reportName), os.ModePerm); err != nil {
 		log.Fatalf("Could not create \"report\" directory: %w", err)
 	}
+
+	// create the work log for go routine workers to use
+	workLog = &namespacesWorkLog{}
 	for _, ns := range namespaces.Items {
+		workLog.namespaces = append(workLog.namespaces, ns.Name)
+	}
 
-		log.Printf("Processing namespace: %s", ns.Name)
-		if shouldExclude(ns.Name, ignoreNamespaces, targetNamespaces) {
-			continue
+	// todo acpana parametrize as a flag
+	log.Printf("Starting worker threads to process namespaces")
+	var wg sync.WaitGroup
+	for i := 0; i < workerRountines; i++ {
+		if workLog.Done() {
+			// if there's no more work stop spawning worker routines;
+			// we're done.
+			break
 		}
 
-		if err := os.Mkdir(filepath.Join(reportName, ns.Name), os.ModePerm); err != nil {
-			log.Printf("Error creating directory for namespace %s:, err: %w; skipping", ns.Name, err)
-			continue
-		}
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
 
-		for _, res := range resources {
-			gvr := schema.GroupVersionResource{
-				Group:    res.Group,
-				Version:  res.Version,
-				Resource: res.Name,
+			if workLog.Done() {
```

Comment:
Can we just rely on GetWork() returning "" ?  Do we need the separate Done method?